pkg/cloud/rgraph/exec/retrieable_action.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024 Google LLC
+
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"context"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+)
+
+// retriableAction is a decorator to the action adding retry mechanism
+type retriableAction struct {
+	Action
+	isRetriable func(error) bool
+}
+
+// NewRetriableAction decorates Action with retry provider function
+func NewRetriableAction(a Action, rp func(error) bool) Action {
+	return &retriableAction{a, rp}
+}
+
+// Run executes Action. If error is returned retry provider checks if Action
+// should be rerun
+func (ra *retriableAction) Run(ctx context.Context, c cloud.Cloud) (EventList, error) {
+	for {
```

Comment:
The infinite loop concerns me. We should probably mandate that the caller supplies a maximum number of retries.

Reply:
added comment